tests/test_sqla.py::TestSQLAlchemy::test_hyperlink_related_field_external fails under Python 3.13

While a rebuild of the flask-marshmallow package in Debian recently it did turn out that the test suite is failing with an error message about a unclosed database connection in a Python 3.13 environment.

The full log relevant part is visible in [Debian Bugreport #1090273](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1090273)

```python
============================= test session starts ==============================
platform linux -- Python 3.13.1, pytest-8.3.4, pluggy-1.5.0
rootdir: /<<PKGBUILDDIR>>/.pybuild/cpython3_3.13_flask_marshmallow/build
configfile: pyproject.toml
plugins: typeguard-4.4.1
collected 49 items

tests/test_core.py ......                                                [ 12%]
tests/test_fields.py .....................                               [ 55%]
tests/test_sqla.py ......E.                                              [ 71%]
tests/test_validate.py ..............                                    [100%]

==================================== ERRORS ====================================
____ ERROR at setup of TestSQLAlchemy.test_hyperlink_related_field_external ____

cls = <class '_pytest.runner.CallInfo'>
func = <function call_and_report.<locals>.<lambda> at 0x7fe05b5d9800>
when = 'setup'
reraise = (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)

    @classmethod
    def from_call(
        cls,
        func: Callable[[], TResult],
        when: Literal["collect", "setup", "call", "teardown"],
        reraise: type[BaseException] | tuple[type[BaseException], ...] | None = None,
    ) -> CallInfo[TResult]:
        """Call func, wrapping the result in a CallInfo.
    
        :param func:
            The function to call. Called without arguments.
        :type func: Callable[[], _pytest.runner.TResult]
        :param when:
            The phase in which the function is called.
        :param reraise:
            Exception or exceptions that shall propagate if raised by the
            function, instead of being wrapped in the CallInfo.
        """
        excinfo = None
        start = timing.time()
        precise_start = timing.perf_counter()
        try:
>           result: TResult | None = func()

/usr/lib/python3/dist-packages/_pytest/runner.py:341: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3/dist-packages/_pytest/runner.py:242: in <lambda>
    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
/usr/lib/python3/dist-packages/pluggy/_hooks.py:513: in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
/usr/lib/python3/dist-packages/pluggy/_manager.py:120: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
/usr/lib/python3/dist-packages/_pytest/unraisableexception.py:90: in pytest_runtest_setup
    yield from unraisable_exception_runtest_hook()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def unraisable_exception_runtest_hook() -> Generator[None]:
        with catch_unraisable_exception() as cm:
            try:
                yield
            finally:
                if cm.unraisable:
                    if cm.unraisable.err_msg is not None:
                        err_msg = cm.unraisable.err_msg
                    else:
                        err_msg = "Exception ignored in"
                    msg = f"{err_msg}: {cm.unraisable.object!r}\n\n"
                    msg += "".join(
                        traceback.format_exception(
                            cm.unraisable.exc_type,
                            cm.unraisable.exc_value,
                            cm.unraisable.exc_traceback,
                        )
                    )
>                   warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
E                   pytest.PytestUnraisableExceptionWarning: Exception ignored in: <sqlite3.Connection object at 0x7fe05b914d60>
E                   
E                   Traceback (most recent call last):
E                     File "/usr/lib/python3/dist-packages/sqlalchemy/orm/mapper.py", line 2142, in _configure_property
E                       if not hasattr(prop, "_is_polymorphic_discriminator"):
E                              ~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E                   ResourceWarning: unclosed database in <sqlite3.Connection object at 0x7fe05b914d60>

/usr/lib/python3/dist-packages/_pytest/unraisableexception.py:85: PytestUnraisableExceptionWarning
=========================== short test summary info ============================
ERROR tests/test_sqla.py::TestSQLAlchemy::test_hyperlink_related_field_external
```

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -1,6 +1,14 @@
 Changelog
 =========
 
+(unreleased)
+************
+
+Bug fixes:
+
+* Fix setting of scoped session on ``SQLAlchemyAuto`` (:pr:`382`).
+  Thanks :user:`galuszkak` for the PR.
+
 1.4.0 (2026-02-04)
 ******************
 

--- a/src/flask_marshmallow/__init__.py
+++ b/src/flask_marshmallow/__init__.py
@@ -134,12 +134,15 @@ class Marshmallow:
         # If using Flask-SQLAlchemy, attach db.session to SQLAlchemySchema
         if has_sqla and "sqlalchemy" in app.extensions:
             db = app.extensions["sqlalchemy"]
-            SQLAlchemySchemaOpts = typing.cast(
-                sqla.SQLAlchemySchemaOpts, self.SQLAlchemySchema.OPTIONS_CLASS
-            )
-            SQLAlchemySchemaOpts.session = db.session
-            SQLAlchemyAutoSchemaOpts = typing.cast(
-                sqla.SQLAlchemyAutoSchemaOpts, self.SQLAlchemySchema.OPTIONS_CLASS
-            )
-            SQLAlchemyAutoSchemaOpts.session = db.session
+            if self.SQLAlchemySchema is not None:
+                SQLAlchemySchemaOpts = typing.cast(
+                    sqla.SQLAlchemySchemaOpts, self.SQLAlchemySchema.OPTIONS_CLASS
+                )
+                SQLAlchemySchemaOpts.session = db.session
+            if self.SQLAlchemyAutoSchema is not None:
+                SQLAlchemyAutoSchemaOpts = typing.cast(
+                    sqla.SQLAlchemyAutoSchemaOpts,
+                    self.SQLAlchemyAutoSchema.OPTIONS_CLASS,
+                )
+                SQLAlchemyAutoSchemaOpts.session = db.session
         app.extensions[EXTENSION_NAME] = self

--- a/tests/test_sqla.py
+++ b/tests/test_sqla.py
@@ -46,6 +46,10 @@ class TestSQLAlchemy:
 
         yield app_
 
+        if "sqlalchemy" in app_.extensions:
+            db = app_.extensions["sqlalchemy"]
+            db.session.remove()
+            db.engine.dispose()
         ctx.pop()
 
     @pytest.fixture

